### ISSUE TITLE
Cache: Pages return Error 500 if page URL has CMS_INDEX_PAGE property as substring

### ISSUE THREAD

**Issue by joseorsini** [dotCMS/core#9678]
Reproduced in demo.dotcms.com (running 3.5) and after upgrade from 2.5.1 to 3.5.1.
Related to https://my.dotcms.com/tickets/detail.dot?id=8581e5c8-df2a-4b17-9300-985e5a5d6246 (Client is running dotCMS version 3.3.2)

Steps to reproduce:
1. Create a test folder in Website Browser.
2. Create a Page that ends with the current value of CMS_INDEX_PAGE property (by default it's "index"). For testing purposes, the page name and URL were the following:
- /services/retire-index (on demo.dotcms.com site)
- /test/finder/finder-index.html (on upgraded data, CMS_INDEX_PAGE value is "index.html").
1. Open a different browser and go to the new created pages.

The problem is here https://github.com/dotCMS/core/blob/master-3.6/src/com/dotmarketing/cache/LiveCache.java#L103-L107 because it tries to substring the current Page URL, which does not match with the specified logic on this code. After calling the page again in Live Mode, it's displayed.

Stacktrace:

```
[29/08/16 09:50:24:838 EDT]  INFO guava.GuavaCache: ***  Building Cache : parentpermissionablecache, size:10000,Concurrency:32
[29/08/16 09:50:32:065 EDT] ERROR velocity.ClientVelocityServlet: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
        at java.lang.String.substring(String.java:1967) ~[?:1.8.0_60]
        at com.dotmarketing.cache.LiveCache.addToLiveAssetToCache(LiveCache.java:103) ~[dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at com.dotmarketing.cache.LiveCache.getPathFromCache(LiveCache.java:301) ~[dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at com.dotmarketing.cache.LiveCache.getPathFromCache(LiveCache.java:179) ~[dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at com.dotmarketing.velocity.VelocityServlet.doLiveMode(VelocityServlet.java:402) ~[dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at com.dotmarketing.velocity.VelocityServlet.service(VelocityServlet.java:242) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:725) [servlet-api.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:721) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:466) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:391) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:318) [catalina.jar:8.0.18]
        at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:231) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at com.dotmarketing.filters.AutoLoginFilter.doFilter(AutoLoginFilter.java:61) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at com.dotmarketing.cms.urlmap.filters.URLMapFilter.doFilter(URLMapFilter.java:307) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:49) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:90) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:33) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:146) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239) [catalina.jar:8.0.18]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206) [catalina.jar:8.0.18]
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219) [catalina.jar:8.0.18]
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106) [catalina.jar:8.0.18]
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:501) [catalina.jar:8.0.18]
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:142) [catalina.jar:8.0.18]
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79) [catalina.jar:8.0.18]
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:610) [catalina.jar:8.0.18]
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88) [catalina.jar:8.0.18]
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:516) [catalina.jar:8.0.18]
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1086) [tomcat-coyote.jar:8.0.18]
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:659) [tomcat-coyote.jar:8.0.18]
        at org.apache.coyote.http11.Http11NioProtocol$Http11ConnectionHandler.process(Http11NioProtocol.java:223) [tomcat-coyote.jar:8.0.18]
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1558) [tomcat-coyote.jar:8.0.18]
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1515) [tomcat-coyote.jar:8.0.18]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_60]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_60]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.0.18]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_60]
```

**Comment by joseorsini** [comment]
PR: https://github.com/dotCMS/core/pull/9680

**Comment by jgambarios** [comment]
@bryanboza please make sure index pages are working properly.

**Comment by bryanboza** [comment]
Fixed, tested on master-3.7 // H2 // FF
